pocketimdb/users/models.py

```diff
@@ -1,7 +1,38 @@
 from django.db import models
-from django.contrib.auth.models import AbstractUser
+from django.contrib.auth.models import AbstractUser, BaseUserManager
+
+class CustomUserManager(BaseUserManager):
+
+    def create_user(self, email, name, password = None, **extra_fields):
+        if not email:
+            raise ValueError('The email must be set')
+        if not name:
+            raise ValueError('The name must be set')
+        email = self.normalize_email(email)
+        user = self.model(email=email, name=name, **extra_fields)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, name, password = None, **extra_fields):
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+
+        if extra_fields.get('is_staff') is not True:
+            raise ValueError('Superuser must have is_staff=True.')
+        if extra_fields.get('is_superuser') is not True:
+            raise ValueError('Superuser must have is_superuser=True.')
+        return self.create_user(email, name, password, **extra_fields)
+
 
 class User(AbstractUser):
-    first_name = models.CharField(max_length=150)
-    last_name = models.CharField(max_length=150)
-    email = models.EmailField()
\ No newline at end of file
+    username = None
+    first_name = None
+    last_name = None
+    email = models.EmailField(max_length=255, unique=True)
+    name = models.CharField(max_length=100, default='')
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = ['name']
```

Comment:
za sta nam sluzi ovaj `REQUIRED_FIELDS ` ?

Reply:
`REQUIRED_FIELDS` se odnosi na polja koja su zahtevana pri kreiranju superuser-a, u sustini nisam taj name morao da stavim ali eto ovim sam podesio da kada se kreira superuser uz mail i password mora i name da mu se zada.

Reply:
mislim da nam je to nepotrebno ovdje, a i to mozes zadati tako sto stavis samo nad tim polje blank=False, null=False

Reply:
<img width="817" alt="Screen Shot 2021-01-14 at 14 31 02" src="https://user-images.githubusercontent.com/44477868/104597175-4a247e00-5675-11eb-9ad1-cd1754cfece1.png">

Zbog ovog sto pise u prvoj recenici sam to ubacio, dakle da kad vrsim komandu `createsuperuser` ponudi name polje da se unese kroz terminal uz email i password.

Reply:
U svakom slucaju mogu izbaciti to nije nesto sto je neophodno, hteo sam vise da vidim kako i to radi. Svakako moram imati to polje sa praznim nizom jer po defaultu je `REQUIRED_FIELDS = ['email']`, a polje koje je `USERNAME_FIELD` ne sme da se nadje u tom nizu, pa moram override-ovati taj niz svakako.

Reply:
aha aha, nista ostavi onda 👍